content-services/content-sources-backend
# Fixes 4549: add more tasks to task cleanup

## Summary
Adds more task types to the tasks that get cleaned up nightly. Now all tasks get cleaned up if they are older than 10 days. The deletion tasks only get cleaned if they completed.

## Testing steps
1. Create a repository and template.
2. Delete a repository and template.
3. This gives a few task types to work with. Update the tasks to change the finished_at times and verify they get cleaned up.

You can use this SQL query to update a task that will always get cleaned up
```
UPDATE tasks
SET queued_at = '2024-08-08 15:24:52.793340 +00:00', started_at = '2024-08-09 15:24:52.793340 +00:00', finished_at = '2024-08-10 15:24:52.793340 +00:00'
WHERE id = '<task id>';
```

You can use this SQL query to update a task that will get cleaned up only if it succeeded. The only difference here is the status is failed.
```
UPDATE tasks
SET queued_at = '2024-08-09 15:24:52.793340 +00:00', started_at = '2024-08-09 15:24:52.793340', finished_at = '2024-08-10 15:24:52.793340 +00:00', status = 'failed'
WHERE id = '<task id>';
```
4. run `go run cmd/external-repos/main.go nightly-jobs`
5. Verify the expected tasks are cleanedup
